pkg/storage/mongodb/repository.go

```diff
@@ -0,0 +1,129 @@
+package mongodb
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/segmentio/ksuid"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/hisshoes/crypto-rebalancer/pkg/portfolio"
+)
+
+// Storage - track portfolios and assets in memory
+type Storage struct {
+	db         *mongo.Client
+	portfolios *mongo.Collection
+	assets     *mongo.Collection
+}
+
+// NewRepository returns an object implementing portfolio repo interface storing in mongodb
+func NewRepository(url string) Storage {
+	// connect to the db and create a client
+	client, err := mongo.Connect(context.Background(), options.Client().ApplyURI(url))
+	if err != nil {
+		log.Fatal("Failed to create mongodb client")
+	}
+
+	// setup the collections required for the repository implementation
+	portfolios := client.Database("cryptos").Collection("portfolios")
+	assets := client.Database("cryptos").Collection("assets")
+
+	return Storage{
+		db:         client,
+		portfolios: portfolios,
+		assets:     assets,
+	}
+}
+
+// GetAssetPrice get the price related to an asset
+func (s Storage) GetAssetPrice(n string) (float64, error) {
+	ctx, c := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
So a common paradigm is that the context is passed into calls as the first parameter. It means the caller can decide what timeout (or not) it wants and then it can also have cancellations/timeouts based on browser/user activity from a frontend.

My recommendation is to add a `ctx context.Context` to the start of all of these calls that are going to happen as a part of the services use (i.e. not just used during startup).